Tests/SonarQube.MSBuild.Tasks.UnitTests/TaskUtilitiesTests.cs

```diff
@@ -0,0 +1,140 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="TaskUtilitiesTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarQube.Common;
+using System.Diagnostics;
+using System.IO;
+using TestUtilities;
+
+namespace SonarQube.MSBuild.Tasks.UnitTests
+{
+    [TestClass]
+    public class TaskUtilitiesTests
+    {
+        public TestContext TestContext { get; set; }
+
+        #region Tests
+
+        [TestMethod]
+        [TestCategory("Task")] // Regression test for bug http://jira.codehaus.org/browse/SONARMSBRU-11
+        public void TaskUtils_LoadConfig_RetryIfConfigLocked_ValueReturned()
```

Comment:
Same two tests as before, but with the duplicated locking part factored out into a new method.